### PR TITLE
Display lines up to a certain number

### PR DESCRIPTION
**Pull request by feuloren**
Small feature proposal :

Introduces two customization:
- `relative-line-numbers-max-count'
- `relative-line-numbers-limit-lines'

The usage for me is to have a margin of minimal width by setting
`relative-line-numbers-max-count' to 9 (or less).

### CODE REVIEW

**Comment by Fanael** [comment]
I'd like it more if it were a single variable that's either `nil` or a number. Can you make it so?

**Comment by feuloren** [comment]
Good idea. I had some problems with customize and nil / integer choice so I made it so that 0 deactivates the limit, is it ok with you ?

**Comment by Fanael** [comment]
Sure.

File: relative-line-numbers.el [hunk 1]
```diff
@@ -47,6 +47,17 @@
   :type 'number
   :group 'relative-line-numbers)
 
+(defcustom relative-line-numbers-max-count 0
+  "Count only up to this number of lines if greater than zero."
+  :type 'integer
+  :group 'relative-line-numbers)
+
+(defcustom relative-line-numbers-max-count 9
```
Comment (Fanael) [inline relative-line-numbers.el]: This defcustom seems reduntant.

File: relative-line-numbers.el [hunk 2]
```diff
@@ -129,8 +140,16 @@ WINDOW is the window to show overlays in."
             (,lineoffsetsym 0)
             (,windowsym ,window))
        (while ,(if (eq direction :forward)
-                   `(and (not (eobp)) (< (point) ,limitsym))
-                 `(and (not (bobp)) (> (point) ,limitsym)))
+                   `(and (not (eobp))
+			 (< (point) ,limitsym)
+			 (if (eq relative-line-numbers-max-count 0)
+			     t
+			     (< ,lineoffsetsym relative-line-numbers-max-count)))
```
Comment (Fanael) [inline relative-line-numbers.el]: Probably should be
```emacs-lisp
(or (eq relative-line-numbers-max-count 0)
    (< ,lineoffsetsym relative-line-numbers-max-count))
```
```

File: relative-line-numbers.el [hunk 3]
```diff
@@ -129,8 +140,16 @@ WINDOW is the window to show overlays in."
...
+                 `(and (not (bobp))
+		       (> (point) ,limitsym)
+		       (if (eq relative-line-numbers-max-count 0)
+			   t
+			   (< (- relative-line-numbers-max-count) ,lineoffsetsym))))
```
Comment (Fanael) [inline relative-line-numbers.el]: Ditto.

File: relative-line-numbers.el (hunk 1 above)
Comment (feuloren) [inline relative-line-numbers.el]: Indeed, fixed

File: relative-line-numbers.el (hunk 3 above)
Comment (feuloren) [inline relative-line-numbers.el]: Good call.

**Comment by Fanael** [comment]
Squashed, untabified and commited in f7ce5d362.

**Pull request closed without merging**